src/ai/ai.cpp

```diff
@@ -104,17 +104,32 @@ void update_ai_general_status(sys::state& state) {
 	}
 }
 
+static void get_alliance_targets_by_adjacency(sys::state& state, dcon::nation_id n, dcon::nation_id adj, std::vector<dcon::nation_id>& alliance_targets) {
+	for(auto nb : state.world.nation_get_nation_adjacency(adj)) {
+		auto other = nb.get_connected_nations(0) != adj ? nb.get_connected_nations(0) : nb.get_connected_nations(1);
+		if(other != n && other.get_is_player_controlled() == false && !(other.get_overlord_as_subject().get_ruler()) && !nations::are_allied(state, n, other) && !military::are_at_war(state, other, n) && ai_will_accept_alliance(state, other, n))
+			alliance_targets.push_back(other.id);
+	}
+}
+
 void form_alliances(sys::state& state) {
 	static std::vector<dcon::nation_id> alliance_targets;
 
 	for(auto n : state.world.in_nation) {
 		if(!n.get_is_player_controlled() && n.get_ai_is_threatened() && !(n.get_overlord_as_subject().get_ruler())) {
 			alliance_targets.clear();
 
-			for(auto nb : n.get_nation_adjacency()) {
-				auto other = nb.get_connected_nations(0) != n ? nb.get_connected_nations(0) : nb.get_connected_nations(1);
-				if(other.get_is_player_controlled() == false && !(other.get_overlord_as_subject().get_ruler())  && !nations::are_allied(state, n, other) && !military::are_at_war(state, other, n) && ai_will_accept_alliance(state, other, n))
-					alliance_targets.push_back(other.id);
+			// Adjacency with us
+			get_alliance_targets_by_adjacency(state, n.id, n.id, alliance_targets);
+			// Adjacency with rival (useful for e.x, Chile allying Paraguay to fight bolivia)
+			get_alliance_targets_by_adjacency(state, n.id, n.get_ai_rival(), alliance_targets);
+			// Adjacency with people who are at war with us
+			for(auto wp : state.world.nation_get_war_participant(n)) {
+				bool as_attacker = wp.get_is_attacker();
+				for(auto p : state.world.war_get_war_participant(wp.get_war())) {
+					if(p.get_is_attacker() == !as_attacker)
+						get_alliance_targets_by_adjacency(state, n.id, p.get_nation(), alliance_targets);
```

Comment:
And this is not a good idea, because alliances are more permanent for the AI while wars are temporary. And the AI shouldn't be forming alliances just to get dragged into a war in any case. Furthermore, just because you are in a war, doesn't mean that you can call allies in anyways (not war leader, or not defending), so most of these alliances will be useless in any case.

Reply:
the AI should take advantage of the situation and not be "exactly" bound to some alliance, we already do ignore diplomatic relations to an extent so why don't we just let the AI don't care about their allies in the first place? this type of behaviour allows the AI to just take advantage instead of being taken advantage of

Reply:
Just trust me on this one for now